xgboost_ray/main.py

```diff
@@ -421,6 +382,8 @@ def _train(params: Dict,
            *args,
            evals=(),
            ray_params: RayParams,
+           _checkpoint: _Checkpoint,
+           _additional_results: Dict,
```

Comment:
hmm, what does this do?

Reply:
Basically we're passing a checkpoint and results reference to be able to record (and return) the contents even in the case of failure. Since we re-raise the actor error in the try/catch block in `_train()`, this seemed like the best way to support keeping checkpoint and results states, and since `_train()` is an internal function it should be fine to use this API here. 
I'm open to suggestions here, though.

Reply:
Got it. So just for my understanding, `additional_results` and `num_boost_rounds` are not correctly being handled during failure in the current implementation correct? Without this PR, when a training run has to restart, each actor loads its local checkpoint, but it starts again from iteration 0 and previous `additional_results` are not being saved, is that right?

Reply:
Yes, that is exactly right. Without this PR the restarted actors train for `num_boost_rounds` again and forfeit any additional results they collected before.